packages/core/src/migration/types.ts

```diff
@@ -0,0 +1,14 @@
+import { DatabasePool } from 'slonik';
+import { z } from 'zod';
+
+export const databaseVersionGuard = z.object({
+  version: z.string(),
+  updatedAt: z.string().optional(),
+});
+
+export type DatabaseVersion = z.infer<typeof databaseVersionGuard>;
+
+export type MigrationScript = {
+  up: (pool: DatabasePool) => Promise<void>;
+  down: (pool: DatabasePool) => Promise<void>;
```

Comment:
nit: not in this PR scope, just might need to consider how should we test these scripts. e.g if someone only provides a detailed up script but. down: async () => {}; Not able to guard it.

Reply:
Good call, but I am not sure yet, created LOG-4213 to track.